astropy/coordinates/sky_coordinate.py

```diff
@@ -1286,13 +1287,25 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
         vcorr : `~astropy.units.Quantity` with velocity units
             The  correction with a positive sign.  I.e., *add* this
             to an observed radial velocity to get the barycentric (or
-            heliocentric) velocity.
+            heliocentric) velocity. If m/s precision or better is needed,
+            see the notes below.
 
         Notes
         -----
-        The algorithm here is sufficient to perform corrections at the ~1 to
-        10 m/s level, but has not been validated at better precision.  Future
-        versions of Astropy will likely aim to improve this.
+        The barycentric correction is calculated to higher precision than the
+        heliocentric correction and includes additional physics (e.g time dilation).
+        Use barycentric corrections if m/s precision is required.
+
+        The algorithm here is sufficient to perform corrections at the mm/s level, but
+        care is needed in application. Strictly speaking, the barycentric correction is
+        multiplicative and should be applied as::
+
+           sc = SkyCoord(1*u.deg, 2*u.deg)
+           vcorr = sc.rv_correction(kind='barycentric', obstime=t, location=loc)
+           rv = rv + vcorr + rv * vcorr / consts.c
```

Comment:
Now that it's actually useful, we should add something to do this automatically.  That's definitely a follow-on PR since it would be a new function/method, but now it's definitely justified.

Reply:
agreed! For 3.0 I'd like to do a PR that includes the effects of the proper motion and distance if they're present on the ```SkyCoord``` so I could add a function/method to that PR